pkg/convert/tf.go

```diff
@@ -1141,10 +1141,68 @@ func convertTemplateWrapExpr(state *convertState,
 	return tokens
 }
 
+func convertTemplateJoinExpr(state *convertState, inBlock bool, scopes *scopes,
+	fullyQualifiedPath string, expr *hclsyntax.TemplateJoinExpr,
+) hclwrite.Tokens {
+	forExpr := expr.Tuple.(*hclsyntax.ForExpr)
+	if forExpr == nil {
+		contract.Failf("Template Join Expression must contain for expression, but contains %T", expr.Tuple)
+	}
```

Comment:
Are all TemplateJoinExprs for-expressions?

Reply:
They are :)